mockito/inorder.py

```diff
@@ -17,11 +17,89 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 # THE SOFTWARE.
+from __future__ import annotations
 
+from collections import Counter, deque
+from typing import List, Any, Tuple, Deque
+
+from .verification import VerificationError
+from .invocation import RealInvocation
+from .mocking import Mock
 from .mockito import verify as verify_main
+from .mock_registry import mock_registry
+from .observer import Observer
 
 
 def verify(object, *args, **kwargs):
     kwargs['inorder'] = True
     return verify_main(object, *args, **kwargs)
 
+
+class InOrder(Observer[Mock]):
+
+    def __init__(self, mocks: List[Any]):
+        """
+        :type mocks: List[Any] = List of mocks.
+        """
+        counter = Counter(mocks)
+        duplicates = [d for d, freq in counter.items() if freq > 1]
+        if duplicates:
+            raise ValueError(
+                f"The following Mocks are duplicated: {duplicates}"
+            )
+        self._mocks = mocks
+
+        for mock in self._mocks:
+            m = mock_registry.mock_for(mock)
+            if m:
+                m.attach(self)
+
+        self.ordered_invocations: Deque[
+            Tuple[Mock, RealInvocation]
+        ] = deque()
+
+    @property
+    def mocks(self):
+        return self._mocks
+
+    def update(self, subject: Mock) -> None:
+        """
+        Observer method that received an
+        invocation notification from the subject.
+
+        :param subject: subject to be added to the list of ordered invocation
+        """
+        self.ordered_invocations.append(
+            (subject, subject.invocations[-1])
```

Comment:
`subject would have been to generic anyway, but subject.invocations[-1] is an implementation detail.  Just pass the invocation in.  self.invocations: deque[RealInvocations]